File: db_api/src/main/java/Post.java

import io.quarkus.mongodb.panache.PanacheMongoEntity;
import io.quarkus.mongodb.panache.PanacheMongoEntityBase;
import io.quarkus.mongodb.panache.common.MongoEntity;
import org.bson.codecs.pojo.annotations.BsonId;
import org.bson.codecs.pojo.annotations.BsonProperty;
import org.bson.types.ObjectId;

import java.util.List;

@MongoEntity(collection = "MySession3566")
public class Post {

    @BsonProperty("_id")
    public Integer _id;

    public String body;
    public List<Comment> comments; // Array of embedded comments
    public int createdBy; // User ID of the creator
    public String date; // Creation date of the post
    public List<String> hashtags; // List of hashtags
    public String hour; // Creation hour

    public Images images; // Nested images object

    public List<Integer> likedBy; // Array of user IDs who liked the post
    public int likes; // Count of likes
    public int shares; // Share count
    public Integer sharedBy; // User ID who shared the post
    public String pseudo; // User's pseudo name
    public boolean showComments; // Boolean flag
}

File: client/src/app/model/post.ts
import {PostComment} from "./post-comment";


export interface Post {

  _id: number; // Numeric _id for the post
  body: string; // Main content of the post
  comments: PostComment[]; // Array of embedded comments
  createdBy: number; // User ID of the creator
  date: string; // Creation date of the post
  hashtags: string[]; // Array of hashtags
  hour: string; // Creation hour
  images: {
    url: string, // Image URL
    title: string // Image title
  };
  likedBy: number[]; // Array of user IDs who liked the post
  likes: number; // Count of likes
  pseudo: string;
  showComments: boolean;
  shares: number;
  sharedBy: number;


}

